### PR TITLE
feat: [CI-5380]: Add tags to VM instances

### PR DESCRIPTION
**Pull request by shubham149**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/36825)

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Failed_** 
 


            
 #### 2 Issues 
* Code issue 1 Code issues 
* Anti-pattern 1 Anti-patterns 

 [Details](https://app.embold.io/organization/gh/harness/repositories/7752dbb3099d2dba3b5d37f5931a5e4c/pull-request/36825)

File: 332-ci-manager/service/src/main/java/io/harness/ci/execution/integrationstage/VmInitializeUtils.java [hunk 1]
```diff
@@ -192,4 +204,50 @@ public OSType getOS(Infrastructure infrastructure) {
...
+      labels.put(STAGE_RUNTIME_ID_ATTR, stageRuntimeID);
+    }
+    if (isLabelAllowed(String.valueOf(buildNumber))) {
+      labels.put(BUILD_NUMBER_ATTR, String.valueOf(buildNumber));
+    }
+    return labels;
+  }
+
+  private boolean isLabelAllowed(String label) {
+    if (label == null) {
```
Comment (vistaarjuneja) [inline 332-ci-manager/service/src/main/java/io/harness/ci/execution/integrationstage/VmInitializeUtils.java:247]: we should also check if label is empty

Comment (shubham149) [reply 332-ci-manager/service/src/main/java/io/harness/ci/execution/integrationstage/VmInitializeUtils.java]: I have updated the code. Labels that we are setting will always be filled. Even if empty label value is set, it is fine.

**Comment by shubham149** [comment]
trigger messagecheck

**Review by raghavharness — APPROVED** [review]
> Reviewed 6 of 6 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/36825)* status: 1. unresolved discussions.

**Review by vistaarjuneja — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/36825#-:-NAOpoSB6eozRvNxuPTv:bnfp4nl)* status: :shipit: complete! ✓

**Review by aman-harness — APPROVED** [review]
:lgtm:

> Reviewed 6 of 6 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/36825#-:-NAOyf723AoennAN3CVs:bnfp4nl)* status: :shipit: complete! ✓

**Pull request merged**